binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -3,11 +3,26 @@
 
 import filecmp
 import os
+import re
 import sys
 from argparse import ArgumentParser
 
-from ..reporter import exception_reporting
 from ._parser import add_version_argument_to
+from ..reporter import exception_reporting
+
+keywords_pattern = re.compile('KEYWORDS="(?P<keywords>.*?)"')
+
+
+def _get_relevant_keywords_set_for(ebuild_content, expected_keywords):
+    try:
+        ebuild_keywords = keywords_pattern.search(ebuild_content).group('keywords')
+    except AttributeError:
```

Comment:
I'll look into that, shouldn't be a big problem to match multiple KEYWORDS. How do you suggest to treat them? Take the union of all KEYWORDS found, or the latest?

Reply:
It seems like multiple `KEYWORDS=` lines are a thing of the past in practice by now, at least in the main Gentoo tree. Here is proof that all ebuilds have one such line at most:

```console
# find /usr/portage/ -type f -not -name \*9999\* -exec fgrep -I KEYWORDS= {} + 2>/dev/null | awk -F: '{print $1}' | sort -u | uniq -c | sort -n | awk '{print $1}' | uniq
1
```

However there are many live ebuilds with `KEYWORDS=` lines that do not actually apply to the containing file, e.g. in `/usr/portage/dev-libs/libmspack/libmspack-9999.ebuild` it reads:
```bash
if [[ ${PV} == "9999" ]] ; then
        EGIT_REPO_URI="https://github.com/kyz/libmspack.git"
        inherit git-r3
        MY_P="${PN}-9999"
else
        KEYWORDS="~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
        MY_PV="${PV/_alpha/alpha}"
        MY_P="${PN}-${MY_PV}"
        SRC_URI="https://www.cabextract.org.uk/libmspack/libmspack-${MY_PV}.tar.gz"
fi
```
So for an easy fix against false positives, we could consider empty keywords when the ebuild filename matches `*9999*.ebuild'. What do you think?

PS: Here is how I found those ebuilds:
```console
# find /usr/portage/ -type f -name \*9999\* -exec grep -Il 'KEYWORDS="[^"]\+"' {} + 2>/dev/null | wc -l
1668
```

Reply:
Taken from https://forums.gentoo.org/viewtopic-t-974234-start-0.html, the "9999" ebuilds are unstable/live ebuilds. My guess is that if anyone is interested in building packages for them, they will build them nightly (or on a regular schedule) rather than when an ebuild changes. If that guess is correct, then they are irrelevant in the output of `gentoo-tree-diff`, so they can always be excluded. Which is achieved by assigning them an empty keywords set, so let's indeed do that.

Reply:
Sounds good. Maybe let's go even further and exclude 9999 ebuilds _before_ keyword checking: I'm thinking of potential future `--keywords='**'` which would still include live ebuilds.